tests/unit/test_instance_connection_manager.py

```diff
@@ -28,7 +29,38 @@
 from google.cloud.sql.connector.utils import generate_keys
 
 
+class BadRefresh(Exception):
+    pass
+
+
+class MockMetadata:
+    def __init__(self, expiration: datetime.datetime) -> None:
+        self.expiration = expiration
+
+
+async def _get_metadata_success(*args: Any, **kwargs: Any) -> MockMetadata:
+    return MockMetadata(datetime.datetime.now() + datetime.timedelta(minutes=10))
+
+
+async def _get_metadata_error(*args: Any, **kwargs: Any) -> None:
+    raise BadRefresh("something went wrong...")
+
+
+async def refresh(icm: InstanceConnectionManager, wait: bool = False) -> asyncio.Task:
```

Comment:
Temporary solution: Will be removed with async interface PR when tests will run with same event loop as ICM and not need `run_coroutine_threadsafe`